x-pack/test/functional/page_objects/gis_page.ts

```diff
@@ -296,10 +296,27 @@ export class GisPageObject extends FtrService {
     };
   }
 
+  // This method is also used by upgrade testing which is not part of PR testing
+  // Please keep in mind when udpating, removing or adding to this method
+  // upgrade needs to be tested too
+  async clearLegendTooltip() {
+    const isTooltipOpen = await this.testSubjects.exists(`layerTocTooltip`, { timeout: 5000 });
+    if (isTooltipOpen) {
+      await this.testSubjects.click(`layerTocTooltip`);
+      await this.common.sleep(1000);
+    }
+  }
+
+  // This method is also used by upgrade testing which is not part of PR testing
+  // Please keep in mind when udpating, removing or adding to this method
+  // upgrade needs to be tested too
   async toggleLayerVisibility(layerName: string) {
-    this.log.debug(`Toggle layer visibility, layer: ${layerName}`);
+    this.log.debug('Inside toggleLayerVisibility');
+    await this.clearLegendTooltip();
     await this.openLayerTocActionsPanel(layerName);
     await this.testSubjects.click('layerVisibilityToggleButton');
+    await this.common.sleep(3000);
```

Comment:
What is this sleep waiting for? Can this be removed?

Reply:
This sleep is waiting after the layer visibility button click to take effect

Reply:
I needed to add the sleep for these for the tests to pass

Reply:
How about using `waitForLayersToLoad` instead of sleep since `waitForLayersToLoad` better codifies when layers are done loading vs static timeout.

Reply:
Added waitForLayersToLoad instead of sleep, pending testing